frontend/public/components/dashboard/dashboards-page/cluster-dashboard/health-item.tsx

```diff
@@ -227,48 +228,45 @@ export const PrometheusHealthItem = withDashboardResources<PrometheusHealthItemP
   },
 );
 
-export const CRDHealthItem = withDashboardResources<CRDHealthItemProps>(
-  ({ watchK8sResource, stopWatchK8sResource, resources, subsystem, models }) => {
-    const modelExists =
-      subsystem.additionalResource && !!models.get(subsystem.additionalResource.kind);
-    React.useEffect(() => {
-      watchK8sResource(subsystem.resource);
-      if (modelExists) {
-        watchK8sResource(subsystem.additionalResource);
-      }
-      return () => {
-        stopWatchK8sResource(subsystem.resource);
-        if (modelExists) {
-          stopWatchK8sResource(subsystem.additionalResource);
-        }
-      };
-    }, [watchK8sResource, stopWatchK8sResource, subsystem, modelExists]);
+export const ResourceHealthItem: React.FC<ResourceHealthItemProps> = ({ subsystem }) => {
+  const { title, resources, healthHandler, popupComponent, popupTitle } = subsystem;
+  const resourcesResult = useK8sWatchResources(resources);
 
-    const resourceResult = resources[subsystem.resource.prop];
-    const additionalResourceResult = subsystem.additionalResource
-      ? resources[subsystem.additionalResource.prop]
-      : null;
+  const nonOptionalResources = Object.keys(resourcesResult).filter(
+    (key) => !resources[key].optional,
+  );
 
-    const healthState = subsystem.healthHandler(resourceResult, additionalResourceResult);
+  const loaded = nonOptionalResources.every((key) => resourcesResult[key].loaded);
```

Comment:
this should be handled by the `healthHandler` IMO. I would just pass the whole result

```
 const healthState = healthHandler(resourcesResult)
```

Reply:
alright moving there